packages/bitget-wallet/src/lib/bitget-wallet.ts

```diff
@@ -0,0 +1,333 @@
+    wallet,
+  };
+};
+
+const BitgetWallet: WalletBehaviourFactory<InjectedWallet> = async ({
+  options,
+  metadata,
+  store,
+  provider,
+  emitter,
+  logger,
+}) => {
+  const _state = setupBitgetWalletState();
+
+  const cleanup = () => {
+    for (const key in _state.wallet.callbacks) {
+      _state.wallet.remove(key);
+    }
+  };
+
+  const signOut = async () => {
+    if (!_state.wallet.isSignedIn()) {
+      return;
+    }
+
+    cleanup();
+    const res = await _state.wallet.signOut();
+
+    if (res === true) {
+      return;
+    }
+
+    const error = new Error(
+      typeof res.error === "string" ? res.error : res.error.type
+    );
+
+    // Prevent signing out by throwing.
+    if (error.message === "User reject") {
+      throw error;
+    }
+
+    // Continue signing out but log the issue.
+    logger.log("Failed to sign out");
+    logger.error(error);
+  };
+
+  const setupEvents = () => {
+    _state.wallet.on("accountChanged", async (newAccountId) => {
+      logger.log("onAccountChange", newAccountId);
+      emitter.emit("signedOut", null);
+    });
+
+    _state.wallet.on("rpcChanged", async (rpc) => {
+      logger.log("onNetworkChange", rpc);
+
+      if (options.network.networkId !== rpc.networkId) {
+        await signOut();
+
+        emitter.emit("signedOut", null);
+        emitter.emit("networkChanged", { networkId: rpc.networkId });
+      }
+    });
+  };
+
+  const getAccounts = async (): Promise<Array<Account>> => {
+    const accountId = _state.wallet.getAccountId() || "";
+    if (!accountId) {
+      return [];
+    }
+
+    // await waitFor(() => !!_state.wallet.account(), { timeout: 100 });
+
+    // const account = _state.wallet.account();
+
+    // When wallet is locked signer is empty an object {}.
+    // if (!account!.connection.signer.getPublicKey) {
+    //   return [{ accountId, publicKey: undefined }];
+    // }
+
+    // debugger
+    const publicKey = await _state.wallet
+      .getPublicKey
+      // account!.accountId,
+      // options.network.networkId
+      ();
+
+    return [
+      {
+        accountId,
+        publicKey: publicKey ? publicKey.toString() : undefined,
+      },
+    ];
+  };
+
+  const isValidActions = (
+    actions: Array<Action>
+  ): actions is Array<FunctionCallAction> => {
+    return actions.every((x) => x.type === "FunctionCall");
+  };
+
+  const transformActions = (actions: Array<Action>) => {
+    const validActions = isValidActions(actions);
+
+    if (!validActions) {
+      throw new Error(
+        `Only 'FunctionCall' actions types are supported by ${metadata.name}`
+      );
+    }
+
+    return actions.map((x) => x.params);
+  };
+
+  const transformTransactions = (
+    transactions: Array<Optional<Transaction, "signerId">>
+  ) => {
+    return transactions.map((transaction) => {
+      return {
+        receiverId: transaction.receiverId,
+        actions: transformActions(transaction.actions),
+        // actions: transaction.actions,
```

Comment:
If it's not needed please remove this commented code.

Reply:
Thank you very much. We have removed the comment code